File: creational-design-patterns/simple-factory-design-pattern/Python/logger.py
from enum import Enum

class LogLevel(Enum):
    INFO = "INFO"
    DEBUG = "DEBUG"
    ERROR = "ERROR"

class ILogger:
    def log(self, msg):
        pass

class DebugLogger(ILogger):
    def log(self, msg):
        print(f"{LogLevel.DEBUG.value}: {msg}")
        
class InfoLogger(ILogger):
    def log(self, msg):
        print(f"{LogLevel.INFO.value}: {msg}")
        
class ErrorLogger(ILogger):
    def log(self, msg):
        print(f"{LogLevel.ERROR.value}: {msg}")    
        
class LoggerFactory:
    @staticmethod
    def create_logger(logLevel):
        if logLevel == LogLevel.DEBUG:
            return DebugLogger() 
        elif logLevel == LogLevel.INFO:
            return InfoLogger() 
        elif logLevel == LogLevel.ERROR:
            return ErrorLogger() 
        else:
            return None
    
def main():
    logger = LoggerFactory.create_logger(LogLevel.ERROR)
    
    logger.log("This is a log message")
    
if __name__ == "__main__":
    main()

File: creational-design-patterns/simple-factory-design-pattern/Typescript/logger.ts
enum LogLevel {
    DEBUG = "DEBUG",
    INFO = "INFO",
    ERROR = "ERROR"
}

class ILogger {
    log(msg: string) {}
}

class DebugLogger extends ILogger {
    log(msg: string){
        console.log(`${LogLevel.DEBUG}: ${msg}`);
    }
}

class InfoLogger extends ILogger {
    log(msg: string){
        console.log(`${LogLevel.INFO}: ${msg}`);
    }
}

class ErrorLogger extends ILogger {
    log(msg: string){
        console.log(`${LogLevel.ERROR}: ${msg}`);
    }
}

class LoggerFactory {
    static createLogger(logLevel: LogLevel) {
        switch(logLevel){
            case LogLevel.DEBUG:
                return new DebugLogger();
            case LogLevel.INFO:
                return new InfoLogger();
            case LogLevel.ERROR:
                return new ErrorLogger();
        }
    }
}

function main() {
    const logger: ILogger = LoggerFactory.createLogger(LogLevel.ERROR);

    logger.log("This is a log message");
}

main()